### ISSUE TITLE
Allow configuration of ts.server.maxFileSize

### ISSUE THREAD

**Issue by pyoor** [microsoft/TypeScript#32420]
## Search Terms
```
configure maxFileSize
largeFileReferenced
```

## Suggestion

Allow the maxFileSize property to be configurable.
## Use Cases

Currently, when trying to import large JSON files, the Typescript Language Service fails to provide intellisense due to the maxFileSize limit being exceeded.

## Examples
```
[Trace  - 10:29:07 AM] <semantic> Event received: largeFileReferenced (0).
Data: {
    "file": "/home/user/project/src/data.json",
    "fileSize": 6534662,
    "maxFileSize": 4194304
}
```
## Checklist

My suggestion meets these guidelines:

* [ ] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [ ] This wouldn't change the runtime behavior of existing JavaScript code
* [ ] This could be implemented without emitting different JS based on the types of the expressions
* [ ] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [ ] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by riverar** [comment]
Yep, just hit this one whilst working with a 43MB JSON file. Was pulling out my hair trying to debug:
> Property 'xxx' does not exist on type '{}'. ts(2339)

Would be great if we could configure this. Or provide an alternate solution (e.g. should I be providing typings instead?)
